frameworks/hdfs/tests/test_sanity.py

```diff
@@ -148,10 +148,23 @@ def test_kill_data_node():
 @pytest.mark.sanity
 @pytest.mark.recovery
 def test_kill_scheduler():
+    foldered_name = sdk_utils.get_foldered_name(config.SERVICE_NAME)
+    scheduler_task_name = sdk_utils.get_task_id_service_name(foldered_name)
+    scheduler_id = sdk_tasks.get_task_ids(foldered_name, scheduler_task_name)
+    journal_ids = sdk_tasks.get_task_ids(foldered_name, "journal")
+    name_ids = sdk_tasks.get_task_ids(foldered_name, "name")
+    data_ids = sdk_tasks.get_task_ids(foldered_name, "data")
+
     sdk_cmd.kill_task_with_pattern(
         "hdfs.scheduler.Main", shakedown.get_service_ips("marathon").pop()
     )
-    config.check_healthy(service_name=sdk_utils.get_foldered_name(config.SERVICE_NAME))
+
+    sdk_marathon.wait_for_deployment_and_app_running(foldered_name, 15 * 60)
+
+    sdk_tasks.check_task_relaunched(scheduler_task_name, scheduler_id)
```

Comment:
Looking at the logs we seem to be getting:
```
6|sdk_tasks-check_task_relaunched(210)|INFO]: Checking (task_name:test.integration.hdfs) (old_task_id:[]) with (ensure_new_task_not_completed:True) is relaunched
```
Where the old-task ID is `[]`. Not sure if this is causing things to fail.